dask/array/reductions.py

```diff
@@ -1000,6 +1019,12 @@ def _nanargmax(x, axis, **kwargs):
 argmax = make_arg_reduction(chunk.max, chunk.argmax)
 nanargmin = make_arg_reduction(chunk.nanmin, _nanargmin, True)
 nanargmax = make_arg_reduction(chunk.nanmax, _nanargmax, True)
+# The above four functions need to be handled by __array_function__
```

Comment:
Could you just use `implements` for this, e.g., something like:
```
nanargmax = implements(np.nanargmax)(make_arg_reduction(chunk.nanmax, _nanargmax, True))
```

Reply:
Yes that works. I found those one-liners a bit too hard to read, but it is shorter and removes the need to import `_HANDLED_FUNCTIONS`. Happy to change it if others prefer this style.

Reply:
okay done. it is a bit cleaner. one may have to read the code a few times to parse that pattern, but nontrivial decorator use is happening in many other places too

Reply:
I actually like @shoyer's suggestion, so that's +1 from me.